### DISCUSSION TITLE
Can I Squeeze More Performance out of my CPU somehow?

### DISCUSSION THREAD

**Discussion by hereforsd5700** [vladmandic/sdnext#1785] [Q&A]
Yes, CPU. 

I managed to generate 2 images at 800x1200 via CPU (Ryzen 7 3700x).  It took a very long time but the system never gave up.  I think I can even push for 960*1280.

On the other hand, in DirectML mode the GPU (5700X) often gives up while generating images bigger than 600x400, and it is a guarantee it will do that if I'm trying do upscale something. It will hang, it will crash, or it will give me BSOD. 

So my question is, are there any tweaks to get more performance out of my CPU? I've given up on running SD on my GPU because it is very unreliable. I'd rather stick to something slow that I know works than gamble with blue screen. 

Thanks.

**Comment by MDMAchine** [comment]
Are you using any tiling options to assist with larger resolutions?
Are you using --medvram or --lowvram?

I find with weaker hardware its better to render at 512 max resolutions and utilize hires (second pass) and/or tiling options to get larger sizes with better VRAM management.

For example for a 9:16 image set width 288 and height 512, then you could set Hires section upscale to 3. Which would be 864x1536. Or you can try Tiled Diffusion with overwrite image size to your parameters. Move tensor to CPU and latent tile batch size lower (1 or 2) for lower spec hardware.

With either option above have Tiled VAE on, uncheck move VAE to GPU and try an encoder size of 1344 and decoder size of 84. (You may even need lower settings)

With these you may still be able to get something out of your GPU.

AFAIK these isnt too much you can do to extract much more from the CPU.

From there you can look into [Ultimate SD Upscaler](https://github.com/Coyote-A/ultimate-upscale-for-automatic1111) for further enlargement. As its pretty gentile on lower spec setups.

**Comment by hereforsd5700** [comment]
Using --medvram.

I need to look into tiled diffusion. I once tried it and it created some horrible, cojoined twin looking abominations.😨

>AFAIK these isnt too much you can do to extract much more from the CPU.

True but I'm impressed how consistent it is on CPU. Even if it takes 40 minutes it will give an output unlike GPU which may or may not run out of memory or give BSOD while generating an image half the size.
